code/game/objects/items/devices/powersink.dm

```diff
@@ -76,14 +76,14 @@
 			src.visible_message("<span class='notice'>[user] activates [src]!</span>")
 			mode = 2
 			icon_state = "powersink1"
-			processing_objects.Add(src)
+			START_PROCESSING(SSobj, src)
 			GLOB.processing_power_items.Add(src)
 		if(2)  //This switch option wasn't originally included. It exists now. --NeoFite
 			src.visible_message("<span class='notice'>[user] deactivates [src]!</span>")
 			mode = 1
 			set_light(0)
 			icon_state = "powersink0"
-			processing_objects.Remove(src)
+			STOP_PROCESSING(SSobj, src)
 			GLOB.processing_power_items.Remove(src)

```

Comment:
I'm still not so sure that `processing_power_items` is a good idea. It's a Bay thing to make sure powersinks process first, supposedly.